cEP-0019.md

```diff
@@ -0,0 +1,208 @@
+generate a meaningful analysis.
+
+#### Ranking list
+
+A ranking list would be displayed on gh-board, according to points
+people get.  A sample formula for points would be as following:
+
+PR = total number of positive reactions (e.g., THUMBS_UP) a person gets
+for all reviews he did.
+NR = total number of negative reactions (e.g., THUMBS_DOWN) a person gets
+for all reviews he did.
+PG = total number of positive reactions a person gives to other people for
+their reviews.
+NG = total number of negative reactions a person gives to other people for
+their reviews.
+
+Note PG and NG are used to encourage people to get involved in this
+reactions feedback system. Also, it may not be limited to PRs. Then final
+score S would be:
+
+S = PR - NR + c1 * PG + c2 * NG
+
+where c1 = 0.5, c2 = 0.8, are coefficients. c2 is slightly larger than c1
+because people are usually reluctant to give negative reactions.
+
+If a person has zeros for all factors, then he is regarded as an inactive
+reviewer and would not be displayed on board.
+
+#### Meta-reviews in Need
+
+Review comments that need meta-review will be displayed on gh-board. There
+are three approaches:
+
+1. The worst case is that the tracking stores in the json which
+PR do not have meta-reviews, and gh-board shows only a list of those PRs
+(and the user must then find the reviews in GitHub PR web app, and do
+meta-review there).
+
+2. An improvement is to store the identifier of the PR comments which
+do not have meta-reviews and gh-board links to the PR review comment
+in GitHub PR web app.
+
+3. In the ultimate version, gh-board would show the review comments.
+In the future, people could do meta-reviews directly on gh-board. This also
+provides an approach to doing meta-reviews anonymously.
+
+#### Other Features
+
+The GSoC project provides a blueprint for future enhancement. Additional
+features could be implemented to make the meta-review system more
+sophisticated:
+
+1. A basic ranking list will be built first. In the best case, we should
+offer an overall ranking list as well as a recent ranking list (user
+could switch to see overall/recent ranking) with sorting option.
+
+2. Offer an option for users to do meta-reviews directly on gh-board. This
+might be a bad idea - people shouldn't do meta-review unless they know the
+whole PR context. This might be a good idea - in the future users can do
+meta-reviews on gh-board anonymously (there is no way for people to do
+anonymous meta-reviews on GitHub page).
+
+3. Display visitor’s own meta-reviews if the visitor logs in. People may want
+to see meta-reviews they have done before.
+
+4. Display visitor's received meta-reviews if the visitor logs in.
+GitHub won't send a notification to a user if he receives any meta-review.
+Suppose I ask for someone's agreement and he uses 👍 to show his agreement.
+But I wouldn't receive any notification. I have to check that page regularly.
```

Comment:
The lack of notifications could be mentioned in the background.

Reply:
A real feature is: gh-board provides notifications of meta-reviews.

This could be done using a per-user atom feed.

Is anyone else providing solutions for this?
Maybe we can implement another tool to solve this?

Reply:
I didn't find any third-party tool for that. The interesting point is that it seems the notification for reactions was open by default before. https://github.com/isaacs/github/issues/631